controller/handler/badge_test.go

```diff
@@ -180,7 +211,68 @@ func TestGetBadgeColorFromUptime(t *testing.T) {
 	}
 }
 
+var (
+	firstCondition  = core.Condition("[STATUS] == 200")
+	secondCondition = core.Condition("[RESPONSE_TIME] < 500")
+	thirdCondition  = core.Condition("[CERTIFICATE_EXPIRATION] < 72h")
+)
+
 func TestGetBadgeColorFromResponseTime(t *testing.T) {
+	defer store.Get().Clear()
+	defer cache.Clear()
+
+	testEndpoint = core.Endpoint{
+		Name:                    "name",
+		Group:                   "group",
+		URL:                     "https://example.org/what/ever",
+		Method:                  "GET",
+		Body:                    "body",
+		Interval:                30 * time.Second,
+		Conditions:              []core.Condition{firstCondition, secondCondition, thirdCondition},
+		Alerts:                  nil,
+		NumberOfFailuresInARow:  0,
+		NumberOfSuccessesInARow: 0,
+	}
+	testSuccessfulResult = core.Result{
+		Hostname:              "example.org",
+		IP:                    "127.0.0.1",
+		HTTPStatus:            200,
+		Errors:                nil,
+		Connected:             true,
+		Success:               true,
+		Timestamp:             timestamp,
+		Duration:              150 * time.Millisecond,
+		CertificateExpiration: 10 * time.Hour,
+		ConditionResults: []*core.ConditionResult{
+			{
+				Condition: "[STATUS] == 200",
+				Success:   true,
+			},
+			{
+				Condition: "[RESPONSE_TIME] < 500",
+				Success:   true,
+			},
+			{
+				Condition: "[CERTIFICATE_EXPIRATION] < 72h",
+				Success:   true,
+			},
+		},
+	}
+	cfg := &config.Config{
+		Metrics: true,
+		Endpoints: []*core.Endpoint{
+			{
+				Name:     "name",
+				Group:    "group",
+				UIConfig: ui.GetDefaultConfig(),
+			},
+		},
+	}
+
+	testEndpoint.UIConfig = ui.GetDefaultConfig()
+	store.Get().Insert(&testEndpoint, &testSuccessfulResult)
+	cfg.Endpoints[0] = &testEndpoint
+
```

Comment:
I'm a little confused here about what you're trying to do. 

Why don't you just set testEndpoint's `UIConfig` field at L234 instead, and then set cfg's `Endpoints` to `testEndpoints`?

Essentially, you'd have something like this:

```go
	cfg := &config.Config{
		Metrics:   true,
		Endpoints: []*core.Endpoint{&testEndpoint},
	}
```
and you wouldn't need to have either L272 and L274

Reply:
I actually didn't know what I did there I only transfered this from the old branch. I implemented your suggestions